src/ngraph/pass/nop_elimination.cpp

```diff
@@ -55,7 +55,7 @@ static bool eliminate_pad(const std::shared_ptr<Node>& node)
 
     if (pad->get_input_shape(0) == pad->get_output_shape(0))
     {
-        return remove_node_update_name(node, node->get_argument(0));
+        return remove_node_update_name(node, node->input_value(0).get_node_shared_ptr());
```

Comment:
Up above, this is for `op::v0::Pad` instead of `op::v1::Pad` The caller of `eliminate_pad` is also the wrong version of `Pad`.

Reply:
made it generic, so this can be called for other cases.